### PR TITLE
add hardware exporter

### PR DESCRIPTION
**Pull request by matthewei**
I have update this deployment files. Please check it!

### CODE REVIEW

File: charts/hardware-exporter/Chart.lock [hunk 1]
```diff
@@ -0,0 +1,6 @@
+dependencies:
+- name: common
+  repository: https://charts.bitnami.com/bitnami
+  version: 2.2.3
+digest: sha256:2c7165542fc01b9e98b577cd8b1095d0ed8267d34b97b6e581a1176bfb8e4dcb
+generated: "2023-02-17T16:30:54.546710642Z"
```
Comment (dongjiang1989) [inline charts/hardware-exporter/Chart.lock:6]: Delete bitnam icommon code dependent

File: charts/hardware-exporter/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+description: Prometheus exporter for hardware and you can use this exporter to get server's bmc and bios' info
+home: https://github.com/kubeservice-stack/hardware-exporter/
+keywords:
+- prometheus
+- hardware-exporter
+- monitoring
+maintainers:
+- email: dlswdjtu@163.com
+  name: shenwei
+name: hardware-exporter
```
Comment (dongjiang1989) [inline charts/hardware-exporter/Chart.yaml:12]: add icon url

File: charts/hardware-exporter/templates/daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
+apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
+kind: DaemonSet
+metadata:
+  name: {{ template "common.names.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
```
Comment (dongjiang1989) [inline charts/hardware-exporter/templates/daemonset.yaml:5]: change `common.names.namespace` with `value.namespaceOverride`
change `common.names.fullname` with `value.fullnameOverride`

and etc.

File: charts/hardware-exporter/values.yaml [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+  metrics: 9696
+
+## Configure Pods Security Context
+## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
+## @param podSecurityContext.enabled Enabled Hardware exporter pods' Security Context
+## @param podSecurityContext.fsGroup Set Hardware exporter pod's Security Context fsGroup
+##
+podSecurityContext:
+  enabled: true
+  fsGroup: 1001
```
Comment (dongjiang1989) [inline charts/hardware-exporter/values.yaml:98]: add ```privileged: true
  seccompProfile:
    type: RuntimeDefault```

**Comment by dongjiang1989** [comment]
@imlense  Please review!

File: charts/hardware-exporter/templates/daemonset.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      containers:
+        - name: hardware-exporter
+          image: {{ template "hardware-exporter.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
+          args:
+            - --web.listen-address=:{{ .Values.containerPorts.metrics }}
+          ports:
```
Comment (imlense) [inline charts/hardware-exporter/templates/daemonset.yaml:62]: Please add liveness & readyness

**Pull request closed without merging**